client_plugin/utils/refcount/refcnt.go

```diff
@@ -508,5 +520,17 @@ func (r *RefCountsMap) updateRefMap() error {
 		log.Debugf("Volume '%s' was found mounted, ref=(%#v)", volName, refInfo)
 	}
 
+	// Add volumes found under mountRoot, reset refMap entry
+	// for those that are present under mountRoot but aren't
+	// mounted.
+	volumes, err := fs.GetMountRootEntries(mountRoot)
+	if err != nil {
+		return err
+	}
+	for _, vol := range volumes {
+		if _, ok := r.refMap[vol]; ok == false {
+			r.refMap[vol] = nil
```

Comment:
I dont remember all refCnt datastructures OTOH. However, this looks inconsistent with overall logic.Why not keep refInfo entry and use refInfo.mounted= false or introduce new field to distinguish volume is attached but not mounted.